manage_breast_screening/auth/oauth.py

```diff
@@ -0,0 +1,51 @@
+from authlib.integrations.django_client import OAuth
+from authlib.jose import JsonWebKey
+from authlib.oauth2.rfc7523 import PrivateKeyJWT
+from cryptography.hazmat.primitives import serialization
+from django.conf import settings
+
+oauth = OAuth()
+
+
+def get_cis2_client():
+    """Return CIS2 OAuth client configured for private_key_jwt."""
+    client = oauth.create_client("cis2")
+    metadata = getattr(client, "server_metadata", None) or {}
+    token_endpoint = metadata.get("token_endpoint")
+    if token_endpoint:
+        try:
+            headers = {}
+            jwk = jwk_from_private_key()
+            kid = jwk.thumbprint() if jwk else None
+            if kid:
+                headers["kid"] = kid
+            client.register_client_auth_method(
+                PrivateKeyJWT(token_endpoint, headers=headers or None)
+            )
+        except Exception:
+            # If already registered or client doesn't support this method, ignore
+            pass
+    return client
+
+
+def jwk_from_private_key():
+    """Build a public JWK from the configured private key PEM (RSA-only).
+
+    Returns:
+        JsonWebKey | None: Public JWK or None on failure.
+    """
+    pem = getattr(settings, "CIS2_PRIVATE_KEY", None)
+    if not pem:
+        return None
+    try:
+        private_key = serialization.load_pem_private_key(
```

Comment:
I was wondering if this could be done with authlib itself, or any of [the built in crypto libraries](https://docs.python.org/3/library/crypto.html) or authlib itself, but it looks like authlib brings in `cryptography` anyway so we may as well use it.